javascript_nodejs/17.proactive-messages/index.js

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const restify = require('restify');
+const path = require('path');
+const ERROR = 1;
+
+// Import required bot services. See https://aka.ms/bot-services to learn more about the different part of a bot.
+const { BotFrameworkAdapter, MemoryStorage } = require('botbuilder');
+const { BotConfiguration } = require('botframework-config');
+
+const MainDialog = require('./dialogs/mainDialog');
+
+// Read botFilePath and botFileSecret from .env file.
+// Note: Ensure you have a .env file and include botFilePath and botFileSecret.
+const ENV_FILE = path.join(__dirname, '.env');
+const env = require('dotenv').config({path: ENV_FILE});
+
+// Create HTTP server.
+let server = restify.createServer();
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`\n${server.name} listening to ${server.url}`);
+    console.log(`\nGet Bot Framework Emulator: https://aka.ms/botframework-emulator`);
+    console.log(`\nTo talk to your bot, open proactive-messages.bot file in the Emulator`);
+});
+
+// .bot file path
+const BOT_FILE = path.join(__dirname, (process.env.botFilePath || ''));
+
+// Read the bot's configuration from a .bot file identified by BOT_FILE.
+// This includes information about the bot's endpoints and Bot Framework configuration.
+let botConfig;
+try {
+    botConfig = BotConfiguration.loadSync(BOT_FILE, process.env.botFileSecret);
+} catch (err) {
+    console.log(`Error reading bot file. Please ensure you have valid botFilePath and botFileSecret set for your environment`);
+    console.log(err);
+    process.exit(ERROR);
+}
+
+// Define the name of the bot, as specified in .bot file.
+// See https://aka.ms/about-bot-file to learn more about .bot file.
+const BOT_CONFIGURATION = 'proactive-messages-bot';
+
+// Load the configuration profile specific to this bot identity.
+const endpointConfig = botConfig.findServiceByNameOrId(BOT_CONFIGURATION);
+
+// Create the adapter. See https://aka.ms/about-bot-adapter to learn more about using information from
+// the .bot file when configuring your adapter.
+const adapter = new BotFrameworkAdapter({
+    appId: endpointConfig.appId || process.env.microsoftAppID,
+    appPassword: endpointConfig.appPassword || process.env.microsoftAppPassword
```

Comment:
Please change to MicrosoftAppId and MicrosoftAppPassword, so that if a developer wanted to drop this code into an already made Azure Bot Service node app, it would run without having to change these variables. (on ABS it's pascal cased)